pkg/controller/servingcert/controller/secret_creating_controller_test.go

```diff
@@ -169,6 +176,99 @@ func TestBasicControllerFlow(t *testing.T) {
 				continue
 			}
 			delete(newSecret.Annotations, api.ServingCertExpiryAnnotation)
+			delete(newSecret.Annotations, api.BetaServingCertExpiryAnnotation)
+			if !reflect.DeepEqual(newSecret.Annotations, expectedSecretAnnotations) {
+				t.Errorf("expected %v, got %v", expectedSecretAnnotations, newSecret.Annotations)
+				continue
+			}
+
+			checkGeneratedCertificate(t, newSecret.Data["tls.crt"], serviceToAdd)
+			foundSecret = true
+
+		case action.Matches("update", "services"):
+			updateService := action.(clientgotesting.UpdateAction)
+			service := updateService.GetObject().(*v1.Service)
+			if !reflect.DeepEqual(service.Annotations, expectedServiceAnnotations) {
+				t.Errorf("expected %v, got %v", expectedServiceAnnotations, service.Annotations)
+				continue
+			}
+			foundServiceUpdate = true
+
+		}
+	}
+
+	if !foundSecret {
+		t.Errorf("secret wasn't created.  Got %v\n", kubeclient.Actions())
+	}
+	if !foundServiceUpdate {
+		t.Errorf("service wasn't updated.  Got %v\n", kubeclient.Actions())
+	}
+}
+
+func TestBasicControllerFlowBetaAnnotation(t *testing.T) {
```

Comment:
Could the gist of these two functions be unified in just one and be run from two different wrapping ones?

Reply:
I think this would need a refactor for each of these specific variant functions, and we will only support the dual annotations for a period of time. I'd like to follow up with that at a later time if we need to. I did reduce the copy-pasta on the other tests, though.

Reply:
Lets' keep it that way, then, no reason to do any major code shoveling to get this PR in :wink: